usb.go

```diff
@@ -165,7 +165,27 @@ func newContextWithImpl(impl libusbIntf) *Context {
 
 // NewContext returns a new Context instance.
 func NewContext() *Context {
-	return newContextWithImpl(libusbImpl{})
+	return ContextOptions{}.New()
+}
+
+// DeviceDiscovery enables a scan of the available USB devices connected to the system
+type DeviceDiscovery int
+
+const (
+	EnableDeviceDiscovery = iota
+	DisableDeviceDiscovery
+)
+
+// ContextOptions holds optional flags checked when creating a new Context
+type ContextOptions struct {
+	DeviceDiscovery DeviceDiscovery
```

Comment:
Document the field here properly, this is the important part that users will be referencing to understand how to initialize the context.

DeviceDiscovery controls USB device discovery. When set to EnableDeviceDiscovery (default), the connected USB devices will be discovered and enumerated, allowing the use os OpenDevices and OpenWithVIDPID. When set to DisableDeviceDiscovery, OpenDevices will not return any devices. OpenDeviceWithFileDescriptor can be used on some systems to open a device that was otherwise discovered through the operating system.

Reply:
what I meant is document the field inside the struct - a user that tries to use ContextOptions will look at the documentation of `ContextOptions` in the first place. So I think this is where it makes sense to add the most verbose documentation. The type itself doesn't need as much of a doc content, although it makes sense to document individual enum values with snippets of the text above.

```
// DeviceDiscovery controls the UBS device discovery and enumeration on Context initialization.
type DeviceDiscovery int

const (
  // EnableDeviceDiscovery means the connected USB devices will be  enumerated 
  // on Context initialization.
  EnableDiscovery DeviceDiscovery = iota
  // DisableDeviceDiscovery means the USB devices are not enumerated and OpenDevices will not
  // return any devices. Without device discovery, OpenDeviceWithFileDescriptor can be used
  // to open devices.
  DisableDiscovery
)
```

Reply:
this is still open